docs: Updates to landing pages and minor edits (#26)
charmed-hpc/docs: README.md

Reviewer: I don't think it's an issue for us to use emojiis here since READMEs are more meant for developers (us and the other Charmed HPC developers) than it is for potential customers. For example, the Rockcraft community uses emojiis sparingly in their community README: https://github.com/rockcrafters

I don't think it hurts to have them in the README, but they should largely be kept out of the actual documentation. _I'm mostly hung up about their removal since it makes the docs repo visually inconsistent with our other repositories._

[![Documentation Status](https://readthedocs.com/projects/canonical-charmed-hpc/badge/?version=latest)](https://canonical-charmed-hpc.readthedocs-hosted.com/en/latest/?badge=latest)
[![Matrix](https://img.shields.io/matrix/ubuntu-hpc%3Amatrix.org?logo=matrix&label=ubuntu-hpc)](https://matrix.to/#/#hpc:ubuntu.com)

Charmed HPC's documentation

For more information on how to access or contribute to Charmed HPC's
documentation, check out the [Getting Started](#-getting-started) and

> [Development](#️-development) sections below.

Reviewer: _I'll give you this one..._ the :point_down: isn't necessary :rofl:

## Getting started

> Periodically, issues within the docs repository will be labeled as 'Good First Issues'. These issues are ideally the first place to get started helping with the Charmed HPC documentation. If there are currently no 'Good First Issues' or none that fit your interest, feel free to reach out on Matrix and ask if the team has any suggestions. Otherwise, small PRs for grammar and formatting are always a good place to start.

Reviewer: Just a couple suggested revisions here. Big thing is changing it from 'Good First Issues' to `good first issue` since that how our new issue labels are styled:

```suggestion
Periodically, issues within the docs repository will be labeled as `good first issue`. These issues are ideally the first place to get started helping with Charmed HPC's documentation. If there are currently no issues labeled `good first issue`, or none of the issues fit your interest, feel free to reach out on [Matrix](https://matrix.to/#/#hpc:ubuntu.com) and ask if the team has any suggestions. Otherwise, small PRs for grammar and formatting are always a good place to start.
```

## Development

> To start the development server and render a local preview run:

Reviewer: [comment]: More succinct; I like it!

For more information on how to access or contribute to Charmed HPC's
documentation, check out the [Getting Started](#-getting-started) and
[Development](#️-development) sections below.

## ✨ Getting started

> Periodically, issues within the docs repository will be labeled as `good first issue`. These issues are ideally the first place to get started helping with Charmed HPC's documentation. If there are currently no issues labeled `good first issue`, or none of the issues fit your interest, feel free to reach out on [Matrix](https://matrix.to/#/#hpc:ubuntu.com) and ask if the team has any suggestions. Otherwise, small PRs for grammar and formatting are always a good place to start.

Reviewer: [nit]: we should likely have `PRs` be fully written out to `pull request` instead. I imagine that the folks we'll get here are more familiar with HPCisms like what HPC means, but not as many folks I find are as familiar with Git-isms. For example, this is what I get when I search "What is a PR":

![image](https://github.com/user-attachments/assets/2432e1ff-fd88-4da2-8996-01aad1484165)

versus "What is a pull request":

![image](https://github.com/user-attachments/assets/4323f5a7-e840-4865-9c19-89fdd91927dd)

```suggestion
Periodically, issues within the docs repository will be labeled as `good first issue`. These issues are ideally the first place to get started helping with Charmed HPC's documentation. If there are currently no issues labeled `good first issue`, or none of the issues fit your interest, feel free to reach out on [Matrix](https://matrix.to/#/#hpc:ubuntu.com) and ask if the team has any suggestions. Otherwise, small pull requests for grammar and formatting are always a good place to start.
```

Author: Ack, agreed.
